ViAn/GUI/projectwidget.cpp

```diff
@@ -804,36 +804,33 @@ void ProjectWidget::context_menu(const QPoint &point) {
  */
 void ProjectWidget::drawing_tag() {
     VideoItem* vid_item;
-    Tag* tag;
-    if (selectedItems().front()->type() == VIDEO_ITEM) {
-        // Create tag drawing
-        vid_item = dynamic_cast<VideoItem*>(selectedItems().front());
-        tag = new Tag();
-        tag->m_name = "Drawing tag";
-        tag->set_drawing_tag(true);
-    } else if (selectedItems().front()->type() == DRAWING_TAG_ITEM) {
+    // Remove the old drawing tag if update was selected
+    if (selectedItems().front()->type() == DRAWING_TAG_ITEM) {
+        vid_item = dynamic_cast<VideoItem*>(selectedItems().front()->parent());
         // Update tag drawing
         DrawingTagItem* item = dynamic_cast<DrawingTagItem*>(selectedItems().front());
-        vid_item = dynamic_cast<VideoItem*>(item->parent());
-        tag = item->get_tag();
-        item->takeChildren();
-        tag->clear_intervals();
+        vid_item->get_video_project()->remove_analysis(item->get_tag());
+        delete selectedItems().front();
+    } else if (selectedItems().front()->type() == VIDEO_ITEM) {
+        vid_item = dynamic_cast<VideoItem*>(selectedItems().front());
     }
+    // Create tag drawing
+    Tag* tag = new Tag();
```

Comment:
Not really necessary and not exactly part of the changes here but I think it would look better if the Tag class constructor took a name parameter as well as a parameter for whether its a drawing or not. I leave it up to you